### PR TITLE
#646 Implement confirm master password

### PR DESCRIPTION
**Pull request by sahalsaad**
Thanks for taking the time to contribute! :gift:

Please follow these steps:

1. work on the right branch: it must be `develop` for new features and `master` for hotfixes
2. make sure the feature you add is not listed in [banned features](https://github.com/keeweb/keeweb/wiki/Unsupported-Features)
3. check compatibility with [the project roadmap](https://github.com/keeweb/keeweb/wiki/TODO) and key features: you should not break anything
4. if your feature is introducing a lot of new UI, or is changing any UX or design concept, please, open an issue to discuss it first
5. please, respect existing style in code, styles and markup (hint: install and enable [editorconfig](http://editorconfig.org/) in your editor)
6. don't add any dependencies
7. test your code: it must work in browser, Windows, OSX and Linux
8. if your code is platform-dependent, please add corresponding switches
9. respect each platform: don't create behaviour which breaks interface guidelines or user expectations 
10. run `grunt` and make sure it's passing 
11. Implement confirm master password #646
screenshot from 2018-08-27 09-48-06
screenshot from 2018-08-27 09-48-22

### CODE REVIEW

File: app/scripts/locales/base.json [hunk 1]
```diff
@@ -428,6 +429,8 @@
   "setFileDontUseKeyFile": "Don't use key file",
   "setFileEmptyPass": "Empty password",
   "setFileEmptyPassBody": "Saving database with an empty password makes it completely unprotected. Do you really want to do it?",
+  "setFilePasswordError": "Error updating password",
+  "setFilePasswordErrorBody": "Master password does not match the confirm master password.",
```
Comment (antelle) [inline app/scripts/locales/base.json]: "confirm master password" should be probably something else, like "confirmed password", or something that make a sentence

File: app/scripts/models/file-model.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ const FileModel = Backbone.Model.extend({
         name: '',
         keyFileName: '',
         passwordLength: 0,
+        confirmPasswordLength: 0,
```
Comment (antelle) [inline app/scripts/models/file-model.js]: Please don't change the model for this feature, it must live in the UI and change the model when both passwords match.

File: app/scripts/views/settings/settings-file-view.js [hunk 3]
```diff
@@ -9,7 +9,6 @@ const Links = require('../../const/links');
 const Format = require('../../util/format');
 const Locale = require('../../util/locale');
 const UrlUtil = require('../../util/url-util');
-const kdbxweb = require('kdbxweb');
```
Comment (antelle) [inline app/scripts/views/settings/settings-file-view.js]: Same as above, I guess, it changes almost everything, please open a new PR then.

File: app/scripts/views/settings/settings-file-view.js [hunk 4]
```diff
@@ -143,6 +146,17 @@ const SettingsFileView = Backbone.View.extend({
             });
             return false;
         }
+
+        if (this.model.get('passwordChanged') && !this.model.get('passwordMatched')) {
+            Alerts.error({
```
Comment (antelle) [inline app/scripts/views/settings/settings-file-view.js]: This should be displayed inline, next to the confirmation field

File: app/scripts/views/settings/settings-file-view.js [hunk 5]
```diff
@@ -382,12 +396,27 @@ const SettingsFileView = Backbone.View.extend({
     blurMasterPass: function(e) {
         if (!e.target.value) {
             this.model.resetPassword();
-            e.target.value = PasswordGenerator.present(this.model.get('passwordLength'));
+            const password = PasswordGenerator.present(this.model.get('passwordLength'));
+            e.target.value = password;
             this.$el.find('.settings__file-master-pass-warning').hide();
+            this.$el.find('#settings__file-confirm-master-pass').val(password);
```
Comment (antelle) [inline app/scripts/views/settings/settings-file-view.js]: Why? The confirmation field must be empty when the focus leaves the master password field, with a message that confirmation is required.

File: app/scripts/views/settings/settings-file-view.js (hunk 4 above)
Comment (sahalsaad) [reply app/scripts/views/settings/settings-file-view.js]: What happen if user click save button if the password entered not match?

File: app/scripts/views/settings/settings-file-view.js (hunk 5 above)
Comment (sahalsaad) [reply app/scripts/views/settings/settings-file-view.js]: Because of this `leave the field blank to use the old password`.

**Pull request closed without merging**